client/question/question.js

```diff
@@ -20,6 +20,79 @@ window.tie = angular.module('tie',
   ['ui.codemirror', 'tieConfig', 'tieData', 'ngCookies']);
 
 /**
+ * String used to label test suites that contain sample input and output
+ * that are presented to the user.
+ *
+ * @type {string}
+ */
+tie.constant('TEST_SUITE_ID_SAMPLE_INPUT', 'SAMPLE_INPUT');
+
+/**
+ * Correctness state for first user attempt to pass a test suite.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_STATE_STARTING',
+  'CORRECTNESS_STATE_STARTING');
+
+/**
+ * Correctness state where the test case input has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_STATE_INPUT_DISPLAYED',
+  'CORRECTNESS_STATE_INPUT_DISPLAYED');
+
+/**
+ * Correctness state where the expeced output has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_STATE_EXPECTED_OUTPUT_DISPLAYED',
+  'CORRECTNESS_STATE_EXPECTED_OUTPUT_DISPLAYED');
+
+/**
+ * Correctness state where the observed output has been displayed to the user.
+ *
+ * @type {string}
+ */
+tie.constant('CORRECTNESS_STATE_OBSERVED_OUTPUT_DISPLAYED',
+  'CORRECTNESS_STATE_OBSERVED_OUTPUT_DISPLAYED');
+
+/**
+ * Object containing variations on feedback text for different scenarios.
+ *
+ * @type {Object.<string, Array.<string>>}
+ */
+tie.constant('CORRECTNESS_FEEDBACK_TEXT', {
+  INPUT_TO_TRY: [
```

Comment:
Re-flagging a previous comment asking if we could have these names match the correctness state constants above (and adding a Karma test to ensure that the keys of CORRECTNESS_FEEDBACK_TEST are exactly those constants).

Reply:
Think this was addressed.